### PR TITLE
Remove trailing blank rows on excel read

### PR DESCRIPTION
**Pull request by AdRiley**
### Pull Request Description

Whenever I record a preppin data for whatever reason their source data always has trailing blank rows which I immediately have to remove with a filter_blank_rows component.

I think we should not read trailing rows that are completely empty of data.

We should still read blank rows that are in the middle of a dataset.

Before

image

After

image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.
- [ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

File: test/Table_Tests/src/IO/Excel_Spec.enso [hunk 1]
```diff
@@ -1304,6 +1304,13 @@ add_specs suite_builder =
...
             table.at "Times" . to_vector . should_equal [Time_Of_Day.new 14 55 52, Time_Of_Day.new 10 30 2, Time_Of_Day.new 23 6 58, Time_Of_Day.new 15 39, Time_Of_Day.new 1 37 21, Time_Of_Day.new 17 16 34, Time_Of_Day.new 10 21 53, Time_Of_Day.new 2 29 9, Time_Of_Day.new 10 20 55, Time_Of_Day.new 2 10 52, Time_Of_Day.new 7 50 24, Time_Of_Day.new 5 46 13, Time_Of_Day.new 18 36 38, Time_Of_Day.new 15 34 42, Time_Of_Day.new 8 25 35]
 
+        group_builder.specify "should remove blank rows from the end of a worksheet" <|
+            ## This worksheet has a header row, a data row and a third blank row. We should not read the blank row from the end.
+            table = (enso_project.data / "WorkbookWithEmptyLastRow.xlsx") . read ..Sheet
```
Comment (GregoryTravis) [inline test/Table_Tests/src/IO/Excel_Spec.enso:1309]: How about a parameter that defaults to remove, but you can switch it to not remove?

**Review by GregoryTravis — APPROVED** [review]

File: test/Table_Tests/src/IO/Excel_Spec.enso (hunk 1 above)
Comment (AdRiley) [reply test/Table_Tests/src/IO/Excel_Spec.enso:1309]: Yeah I thought about that. But is there really a usecase for it? I guess that is a good compromise if we can imagine some edge case where someone wants the empty rows.

**Comment by AdRiley** [comment]
Quick sanity check that this doesn't degrade performance for large excel files

image

Before

image

After

image

**Review by jdunkerley — APPROVED** [review]

File: test/Table_Tests/src/IO/Excel_Spec.enso (hunk 1 above)
Comment (GregoryTravis) [reply test/Table_Tests/src/IO/Excel_Spec.enso:1309]: The edge case is if your expecting the dimensions to exactly match what was in the file, but I think it's rare so fine to leave it out until someone asks for it.

**Review by GregoryTravis — APPROVED** [review]
